mlflow/models/evaluation/default_evaluator.py

```diff
@@ -263,75 +265,163 @@ def plot_feature_importance():
+
+            per_class_metrics.update(self._get_per_class_metrics(y_is_positive, y_pred_is_positive))
+
+            if self.y_probs is not None:
+                fpr, tpr, thresholds = sk_metrics.roc_curve(y_is_positive, prob_of_positive)
+                per_class_roc_curve_data_list.append(
+                    PerClassRocCurveData(postive_class, fpr, tpr, thresholds)
+                )
+                roc_auc = sk_metrics.auc(fpr, tpr)
+                per_class_metrics["roc_auc"] = roc_auc
+
+                precision, recall, thresholds = \
+                    sk_metrics.precision_recall_curve(y_is_positive, prob_of_positive)
+                thresholds = np.append(thresholds, [1.0], axis=0)
+                per_class_precision_recall_curve_data_list.append(
+                    PerClassPrecisionRecallCurveData(postive_class, precision, recall, thresholds)
+                )
+                pr_auc = sk_metrics.auc(recall, precision)
+                per_class_metrics["precision_recall_auc"] = pr_auc
+
+        per_class_metrics_pandas_df = pd.DataFrame(per_class_metrics_list)
+        self._log_pandas_df_artifact(per_class_metrics_pandas_df, "per_class_metrics_data")
+
+        if self.y_probs is not None:
+            per_class_roc_curve_pandas_df = pd.concat(
+                [pd.DataFrame(item._asdict()) for item in per_class_roc_curve_data_list],
+                ignore_index=True
+            )
+            self._log_pandas_df_artifact(per_class_roc_curve_pandas_df, "per_class_roc_curve_data")
+
+            per_class_precision_recall_curve_pandas_df = pd.concat(
+                [pd.DataFrame(item._asdict()) for item in per_class_precision_recall_curve_data_list],
+                ignore_index=True
+            )
+            self._log_pandas_df_artifact(
+                per_class_precision_recall_curve_pandas_df,
+                "per_class_precision_recall_curve_data"
+            )
+
+            def plot_roc_curve():
+                data_series = {
+                    f'Positive Class = {postive_class}': (fpr, tpr)
+                    for postive_class, fpr, tpr, _ in per_class_roc_curve_data_list
+                }
+                plot_lines(
+                    data_series, xlabel='False Positive Rate', ylabel='True Positive Rate',
+                    legend_loc='lower right',
+                    line_kwargs={"drawstyle": "steps-post"}
                 )
 
+            def plot_precision_recall_curve():
+                data_series = {
+                    f'Positive Class = {postive_class}': (recall, precision)
+                    for postive_class, precision, recall, _ in per_class_precision_recall_curve_data_list
+                }
+                plot_lines(
+                    data_series, xlabel='recall', ylabel='precision',
+                    legend_loc='lower left',
+                    line_kwargs={"drawstyle": "steps-post"}
+                )
+
+            if self.num_classes <= 10:
```

Comment:
Solid barrier here. Out of curiosity, what was the count where the plot became hard to follow?

Reply:
Not sure the count, but for 10 classes, each class line contains 50000 points, it still works fine.

Reply:
Just tested for 10 classes, each class line contains 500000 points, plotting still works fine:
![image](https://user-images.githubusercontent.com/19235986/147618669-5a147b51-7b5f-4800-b75a-fb506e688267.png)

Reply:
I think we don't need care too much about the performance of plotting, the limitation may be mainly in javascript plots issue.

Reply:
Q:
Shall we also skip logging ROC/PR curve data as CSV when num_classes > 10 ?
When num_classes is large, the CSV data size will also explode.

Reply:
I add config `max_num_classes_threshold_logging_roc_pr_curve_for_multiclass_classifier` to control this. and, both the curve plot and curve data CSV logging are controlled by the config.